pouta_blueprints/views/stats.py

```diff
@@ -0,0 +1,80 @@
+from flask.ext.restful import marshal_with, fields
+from flask import g
+from flask import Blueprint as FlaskBlueprint
+
+import logging
+
+from pouta_blueprints.models import Blueprint, Instance, User
+from pouta_blueprints.server import restful
+from pouta_blueprints.views.commons import auth
+from pouta_blueprints.utils import requires_admin, memoize
+
+from collections import defaultdict
+
+
+stats = FlaskBlueprint('stats', __name__)
+
+
+def query_blueprint(blueprint_id):
+    return Blueprint.query.filter_by(id=blueprint_id).first()
+
+
+def query_user(user_id):
+    return User.query.filter_by(id=user_id).first()
+
+blueprint_result_fields = {
+
+    'blueprint': {
+        'name': fields.String,
+        'users': fields.Integer,
+        'launched_instances': fields.Integer,
+        'running_instances': fields.Integer,
+    },
+    'overall_running_instances': fields.Integer
+}
+
+
+class StatsList(restful.Resource):
+    @auth.login_required
+    @requires_admin
+    @marshal_with(blueprint_result_fields)
+    def get(self):
+        user = g.user
+        if user.is_admin:
+            instances = Instance.query.all()
+            overall_running_instances = Instance.query.filter(Instance.state != Instance.STATE_DELETED).count()
+
+        get_blueprint = memoize(query_blueprint)
+        get_user = memoize(query_user)
+        per_blueprint_results = defaultdict(lambda: {'users': 0, 'launched_instances': 0, 'running_instances': 0})
+        unique_users = defaultdict(set)
+
+        for instance in instances:
+
+            user = get_user(instance.user_id)
```

Comment:
no need to query user with get_user, instance.user_id should be enough to identify unique users